src/nvim/highlight_group.c

```diff
@@ -1569,7 +1569,16 @@ Dictionary ns_get_hl_defs(NS ns_id, Dict(get_highlight) *opts, Arena *arena, Err
   Boolean link = GET_BOOL_OR_TRUE(opts, get_highlight, link);
   int id = -1;
   if (HAS_KEY(opts, get_highlight, name)) {
-    id = syn_check_group(opts->name.data, opts->name.size);
+    Boolean create = GET_BOOL_OR_TRUE(opts, get_highlight, create);
+    if (HAS_KEY(opts, get_highlight, create)) {
+      create = opts->create;
+    }
```

Comment:
this is redundant and can be removed (`GET_BOOL_OR_TRUE` does everything)

Reply:
done.